library/core/src/result.rs

```diff
@@ -1646,7 +1647,8 @@ impl<T, E> ops::TryV2 for Result<T, E> {
 }
 
 #[unstable(feature = "try_trait_v2", issue = "84277")]
-impl<T, E, F: From<E>> ops::FromResidual<Result<convert::Infallible, E>> for Result<T, F> {
+#[rustc_const_unstable(feature = "const_identity_convert", issue = "none")]
+impl<T, E, F: From<E>> const ops::FromResidual<Result<convert::Infallible, E>> for Result<T, F> {
     #[inline]
     fn from_residual(residual: Result<convert::Infallible, E>) -> Self {
         match residual {
```

Comment:
From my reading of (not yet accepted) rust-lang/rfcs#2632 - [[rendered]](https://github.com/oli-obk/rfcs/blob/const_generic_const_fn_bounds/text/0000-const-generic-const-fn-bounds.md#runtime-uses-dont-have-const-restrictions), my understanding is that this should be fine since we will only implement `ops::FromResidual` constly for types where `F` implements `const From<E>`. For types `F` which only implement `notconst From<E>`, `ops::FromResidual` will be implemented non constly.


```rust
error[E0015]: calls in constant functions are limited to constant functions, tuple structs and tuple variants
    --> library/core/src/result.rs:1655:27
     |
1655 |             Err(e) => Err(From::from(e)),
     |                           ^^^^^^^^^^^^^

error: aborting due to previous error
```

Is my understanding wrong or is this simply a case of `const_trait_impl` being `incomplete_features`? :)

Reply:
> is this simply a case of const_trait_impl being incomplete_features?

This is the problem here. The trivial example https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=a7673524e39afe4cedd1f58e6cc16716 does not even work, so... I guess now that I'm back I should look into this

Reply:
> [...] so... I guess now that I'm back I should look into this

May I ask if there has been any progress? :) And if not, is there anything not too complicated that I could do to help?

Reply:
Sorry, got swallowed by a type alias impl trait hole.

@fee1-dead is on top of all of the impl const trait stuff

Reply:
Hi, sorry for this, I am unaware of the discussion taking place here. For the generic bounds on the impl, it has been fixed by #87273 but we need to wait for #87375 to land before anything happening in libstd. Also have your eyes on #86808, as that is where the libs/lang team will make the decision about const impl stuff in libstd.

Reply:
No worries, thanks for the heads up :)

Reply:
Guess the error should be gone after #87622 then :)